src/subs/WCSFoFRoutine.cpp

```diff
@@ -346,8 +346,17 @@ void FoFClass::sortMatches(int fieldNumber, int minMatches, bool allowSelfMatche
         if (pcat.empty()) continue;
 
         long matches = 0;
+
+        // Sort the matches by their ra position to ensure deterministic behavior downstream.
+        const auto matchSort = [](fof::Match<Point, 2> *a, fof::Match<Point, 2> *b) {
+            return a->getAverage()[0] < b->getAverage()[0];
+        };
+        std::vector<fof::Match<Point, 2> *> matchVector(pcat.size());
+        matchVector.assign(pcat.begin(), pcat.end());
```

Comment:
@Can you note why you need to copy `pcat` here? I'm guessing it's because it's a const ref to a container? If it were mutable, presumably you could just `std::sort` it directly, no?

Reply:
I am copying it because `pcat` is an instance of a child-class of `set` and you can't re-sort a `set` (and I don't want to impose this sorting on the class in general). I copied it into a `vector` because the requirements for the comparison function for `std::sort<vector>` are simpler than for a `set`.